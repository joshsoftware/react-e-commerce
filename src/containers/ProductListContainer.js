import React, { useEffect, useState } from 'react';
import ProductRowContainer from './ProductRowContainer';
import { useDispatch, useSelector } from 'react-redux';
import { getProductList } from '../actions/productListActions';
import ContainerWrapper from '../components/ContainerWrapper';
const ProductListContainer = () => {
  const [loading, setLoading] = useState(false);
  const [page, setPage] = useState(1);
  const { totalPages } = useSelector((state) => state.productListReducer);
  const handleScroll = () => {
    if (
      window.innerHeight + document.documentElement.scrollTop !==
        document.documentElement.offsetHeight ||
      loading
    )
      return;
    console.log(page);
    if (page + 1 <= totalPages || totalPages === 0) {
      setLoading(true);
    }
  };
  useEffect(() => {
    setLoading(true);
    window.addEventListener('scroll', handleScroll);
    return () => window.removeEventListener('scroll', handleScroll);
  }, []);
  const dispatch = useDispatch();
  const changeStates = async () => {
    setTimeout(() => {
      setPage(page + 1);
      setLoading(false);
    }, 2000);
  };
  useEffect(() => {
    if (loading === true) {
      console.log('Page :', page);
      dispatch(getProductList(page));
      changeStates();
    }
  }, [loading]);
  const { productList } = useSelector((state) => state.productListReducer);
  let arr = [];
  for (let i = 0; i < productList.length; i += 3) {
    arr.push(
      <ProductRowContainer key={productList[i].id} products={productList.slice(i, i + 3)} />
    );
  }
  return <ContainerWrapper data={arr} fluid={true} />;
};
export default ProductListContainer;