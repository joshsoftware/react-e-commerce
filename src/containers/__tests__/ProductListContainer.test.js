import React from 'react';
import { render } from '@testing-library/react';
import ProductListContainer from '../ProductListContainer';

describe('ProductList Container', () => {
  it('Must return products list', () => {
    const { asFragment } = render(<ProductListContainer />);
    expect(asFragment()).toMatchSnapshot();
  });
});
