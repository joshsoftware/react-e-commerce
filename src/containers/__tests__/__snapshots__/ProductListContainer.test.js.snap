// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList Container Must return products list 1`] = `
<DocumentFragment>
  <div
    class="container-fluid"
  >
    <div
      class="row"
    >
      <div
        class="col"
      >
        <div
          class="card"
        >
          <img
            alt="Product Image"
            class="card-img"
            src="https://cdn.shopify.com/s/files/1/0011/7312/7180/products/jpeg-jpeg01-t-shirt-tshirt-clothing-black-cyberpunk-goth-geek-cult-23-keep-it-url-1.jpg?v=1581498404"
          />
          <div
            class="card-body"
          >
            <div
              class="card-title"
            >
              Football
            </div>
            <p
              class="card-text"
            >
              $49.99
            </p>
          </div>
        </div>
      </div>
      <div
        class="col"
      >
        <div
          class="card"
        >
          <img
            alt="Product Image"
            class="card-img"
            src="https://cdn.shopify.com/s/files/1/0011/7312/7180/products/jpeg-jpeg01-t-shirt-tshirt-clothing-black-cyberpunk-goth-geek-cult-23-keep-it-url-1.jpg?v=1581498404"
          />
          <div
            class="card-body"
          >
            <div
              class="card-title"
            >
              Baseball
            </div>
            <p
              class="card-text"
            >
              $9.99
            </p>
          </div>
        </div>
      </div>
      <div
        class="col"
      >
        <div
          class="card"
        >
          <img
            alt="Product Image"
            class="card-img"
            src="https://cdn.shopify.com/s/files/1/0011/7312/7180/products/jpeg-jpeg01-t-shirt-tshirt-clothing-black-cyberpunk-goth-geek-cult-23-keep-it-url-1.jpg?v=1581498404"
          />
          <div
            class="card-body"
          >
            <div
              class="card-title"
            >
              Basketball
            </div>
            <p
              class="card-text"
            >
              $29.99
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <div
          class="card"
        >
          <img
            alt="Product Image"
            class="card-img"
            src="https://cdn.shopify.com/s/files/1/0011/7312/7180/products/jpeg-jpeg01-t-shirt-tshirt-clothing-black-cyberpunk-goth-geek-cult-23-keep-it-url-1.jpg?v=1581498404"
          />
          <div
            class="card-body"
          >
            <div
              class="card-title"
            >
              iPod Touch
            </div>
            <p
              class="card-text"
            >
              $99.99
            </p>
          </div>
        </div>
      </div>
      <div
        class="col"
      >
        <div
          class="card"
        >
          <img
            alt="Product Image"
            class="card-img"
            src="https://cdn.shopify.com/s/files/1/0011/7312/7180/products/jpeg-jpeg01-t-shirt-tshirt-clothing-black-cyberpunk-goth-geek-cult-23-keep-it-url-1.jpg?v=1581498404"
          />
          <div
            class="card-body"
          >
            <div
              class="card-title"
            >
              iPhone 5
            </div>
            <p
              class="card-text"
            >
              $399.99
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
